test/common/controllers/base-address.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+
+const proxyquire = require('proxyquire');
+class StubBaseController {}
+
+StubBaseController.prototype.locals = sinon.stub();
+
+const BaseAddressController = proxyquire('../../../apps/common/controllers/base-address', {
+  './base': StubBaseController
+});
+
+describe('BaseAddressController', function() {
+
+  it('extends base-controller', ()=> {
+    BaseAddressController.prototype.should.be.an.instanceOf(StubBaseController);
+  });
+
+  it('has a locals, translateCategories, mapAddress & hasCategories methods', ()=> {
+    BaseAddressController.prototype.should.have.property('locals').that.is.a('function');
+    BaseAddressController.prototype.should.have.property('translateCategories').that.is.a('function');
+    BaseAddressController.prototype.should.have.property('mapAddress').that.is.a('function');
+    BaseAddressController.prototype.should.have.property('hasCategories').that.is.a('function');
+  });
+
+  describe('instance', ()=> {
+    let baseAddressController;
+    const options = {
+      addressKey: 'foo'
+    };
+
+    beforeEach(()=> {
+      StubBaseController.prototype.locals = sinon.stub();
+      baseAddressController = new BaseAddressController(options);
+      baseAddressController.options = options;
+    });
+
+      describe('locals', () => {
+        let req = {
+          sessionModel: {
+            get: sinon.stub()
+          }
+        };
+        let res = {};
+
+        beforeEach(function() {
+          this.stub(BaseAddressController.prototype, 'mapAddress').returns([]);
+          this.stub(BaseAddressController.prototype, 'hasCategories').returns(false);
+        });
+
+        it('returns an empty address list if no addresses exist on the session', () => {
+          const expectedVal = {
+            hasAddresses: false,
+            hasCategories: false,
+            items: []
+          };
+          baseAddressController.locals(req, res).should.deep.equal(expectedVal);
+        });
+
+        it('calls base locals', () => {
+          StubBaseController.prototype.locals.should.not.have.been.called;
+
+          baseAddressController.locals(req, res);
+
+          StubBaseController.prototype.locals.should.have.been.called;
```

Comment:
I prefer to be explicit with the expected amount of calls - use calledOnce, calledTwice etc